libsplinter/src/admin/service/shared.rs

```diff
@@ -2892,14 +2892,40 @@ impl AdminServiceShared {
     }
 
     fn validate_circuit_vote(
-        &self,
+        &mut self,
         proposal_vote: &CircuitProposalVote,
         signer_public_key: &[u8],
         circuit_proposal: &StoreProposal,
         node_id: &str,
         protocol: u32,
     ) -> Result<(), AdminSharedError> {
-        self.validate_protocol_vote(circuit_proposal, protocol)?;
+        if let Err(err) = self.validate_protocol_vote(circuit_proposal, protocol) {
+            warn!(
+                "Removing proposal for {}, no longer valid for agreed protocol version: {}",
+                circuit_proposal.circuit_id(),
+                protocol
+            );
+
+            self.remove_proposal(circuit_proposal.circuit_id())?;
+            self.peers_to_be_removed.push((
```

Comment:
Why remove these here automatically? I don't think this would be expected behavior from a user perspective.